File: chatbot-directline-api/php_demo/step1_generate_token.php
<?php
/**
 * Step 1: Generate access token using Bot Secret
 */

require_once 'config.php';

function generateToken() {
    echo "Step 1: Generating access token...\n";
    
    // Ensure data directory exists
    $dataDir = 'data';
    if (!is_dir($dataDir)) {
        mkdir($dataDir, 0755, true);
    }
    
    $url = 'https://directline.botframework.com/v3/directline/tokens/generate';
    $headers = [
        'Authorization: Bearer ' . BOT_SECRET,
        'Content-Type: application/json'
    ];
    
    // Prepare cURL request
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, $url);
    curl_setopt($ch, CURLOPT_POST, true);
    curl_setopt($ch, CURLOPT_POSTFIELDS, '{}');
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, true);
    
    // Send request
    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);
    
    if ($httpCode === 200) {
        $data = json_decode($response, true);
        $token = $data['token'] ?? null;
        $conversationId = $data['conversationId'] ?? null;
        
        if ($token && $conversationId) {
            echo "✅ Token generated successfully!\n";
            echo "   Conversation ID: {$conversationId}\n";
            echo "   Token expires in: {$data['expires_in']} seconds\n";
            
            // Save token and conversation ID for next steps
            file_put_contents($dataDir . '/token.txt', $token);
            file_put_contents($dataDir . '/conversation_id.txt', $conversationId);
            
            return [$token, $conversationId];
        }
    }
    
    echo "❌ Failed to generate token: HTTP {$httpCode}\n";
    if ($response) {
        echo "Response: {$response}\n";
    }
    
    return [null, null];
}

// Run if called directly
if (basename(__FILE__) === basename($_SERVER['SCRIPT_NAME'])) {
    generateToken();
}
?> 

File: chatbot-directline-api/python_demo/step1_generate_token.py
import requests
import os
from config import BOT_SECRET


def generate_token():
    """Step 1: Generate access token using Bot Secret"""
    print("Step 1: Generating access token...")

    # Get the directory where this script is located
    script_dir = os.path.dirname(os.path.abspath(__file__))
    data_dir = os.path.join(script_dir, "data")
    os.makedirs(data_dir, exist_ok=True)

    url = "https://directline.botframework.com/v3/directline/tokens/generate"
    headers = {
        "Authorization": f"Bearer {BOT_SECRET}",
        "Content-Type": "application/json",
    }

    # Send request to generate token
    response = requests.post(url, headers=headers, json={})

    if response.status_code == 200:
        data = response.json()
        token = data.get("token")
        conversation_id = data.get("conversationId")

        print(f"✅ Token generated successfully!")
        print(f"   Conversation ID: {conversation_id}")
        print(f"   Token expires in: {data.get('expires_in')} seconds")

        # Save token and conversation ID for next steps
        with open(os.path.join(data_dir, "token.txt"), "w") as f:
            f.write(token)
        with open(os.path.join(data_dir, "conversation_id.txt"), "w") as f:
            f.write(conversation_id)

        return token, conversation_id
    else:
        print(f"❌ Failed to generate token: {response.text}")
        return None, None


if __name__ == "__main__":
    generate_token()
